eng/pipelines/templates/Build_and_UnitTest.yml

```diff
@@ -86,8 +86,10 @@ steps:
     scriptType: "inlineScript"
     inlineScript: |
       git remote add github https://github.com/NuGet/NuGet.Client.git
+      # If the CI agent is reused, and the github remote already exists, git will exit with code 3, as per their docs
+      if ($LASTEXITCODE -eq 3) { $LASTEXITCODE = 0 }
```

Comment:
When I tried "git remote add github https://github.com/NuGet/NuGet.Client.git" locally, it looks like the `$LASTEXITCODE` is 128. Not sure if there is anything wrong in my repro.

Reply:
Using the version dropdown at the top of the [docs page](https://git-scm.com/docs/git-remote/2.30.0), it looks like it was added in 2.30.0, in December 2020, so a very new feature. Looking at MicroBuild's software inventory documentation, it looks like they still install git 2.19.2 :(

This script is going to need to be much more complex.

Reply:
I see.. The doc says it will return 3. But it actually returns  [128 ](https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=4497184&view=logs&j=2ea8c801-c4c3-578e-0ff0-bcf4c12d9404&t=267f3845-9303-5f42-91dd-0debfbfbf3ed&l=12)  for version 2.19.2 (both my local machine and CI).
So it means different versions of git return different exitcode? Oh that's not good.

Reply:
Can we remove the remote first and then add it again so that `LASTEXITCODE` will be `0` as suggested [here](https://docs.github.com/en/github/using-git/adding-a-remote#remote-name-already-exists)? 

https://docs.github.com/en/github/using-git/removing-a-remote

Reply:
If the remote name doesn't exist, the remove command will fail as well. 
Will it fail the task if not setting exitcode back to 0?

Reply:
In the latest commit, `git remote remove` command result is ignored and `$LASTEXITCODE` is fetched after `git add` command which should always be `0`.